### PR TITLE
bpo-45917: Add math.exp2() method - return 2 raised to the power of x

### PR DESCRIPTION
**Pull request by Turreted**
https://bugs.python.org/issue45917

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@Turreted

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Review by tim-one — APPROVED** [review]
Looks good - nice work! 😃

File: Lib/test/test_math.py
Comment (mdickinson) [inline Lib/test/test_math.py:508]: Can we add at least one test for a non-integral finite input? E.g., 
```suggestion
        self.ftest('exp2(1)', math.exp2(1), 2)
        self.ftest('exp2(2.3)', math.exp2(2.3), 4.924577653379665)
```

File: Misc/NEWS.d/next/Library/2021-11-28-22-44-54.bpo-45917.lNn_gz.rst
Comment (mdickinson) [inline Misc/NEWS.d/next/Library/2021-11-28-22-44-54.bpo-45917.lNn_gz.rst]: Should this file be deleted? (It looks as though we've ended up with two separate news entries.)

**Review by mdickinson — APPROVED** [review]
LGTM. One suggestion for an extra unit test, and it looks as though one of the added `Misc/NEWS.d` files needs to be deleted.

I'll fire off a round of buildbot checks.

File: Misc/NEWS.d/next/Library/2021-11-28-22-44-54.bpo-45917.lNn_gz.rst
Comment (Turreted) [reply Misc/NEWS.d/next/Library/2021-11-28-22-44-54.bpo-45917.lNn_gz.rst]: 👍

File: Lib/test/test_math.py
Comment (Turreted) [reply Lib/test/test_math.py:508]: Done. It might also be a good idea to add non-integral testcases to other methods, like math.exp, but that's a separate issue

**Comment by mdickinson** [comment]
The buildbots all seem to be either happy, stalled, or failing for reasons that are clearly unrelated to this PR; no evidence of any issues.

Still LGTM! Will merge when CI completes.

File: Doc/library/math.rst
Comment (mdickinson) [inline Doc/library/math.rst:363]: One last nitpick-level suggestion: we're separating the various functions from each other by two blank lines in this file. I have no idea why, but let's keep the consistency.

I don't want to hold this PR up for such a trivial change, so I'll self-merge this change before merging the PR (provided the continuous integration is happy, of course).

```suggestion

.. function:: exp2(x)

   Return *2* raised to the power *x*.

   .. versionadded:: 3.11

```

**Pull request merged**
